store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreInputChainTest.java

```diff
@@ -0,0 +1,131 @@
+import org.junit.jupiter.api.Test;
+
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.federatedstore.operation.AddGraph;
+import uk.gov.gchq.gaffer.federatedstore.operation.FederatedOperation;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.DiscardOutput;
+import uk.gov.gchq.gaffer.operation.impl.GetVariable;
+import uk.gov.gchq.gaffer.operation.impl.If;
+import uk.gov.gchq.gaffer.operation.impl.Map;
+import uk.gov.gchq.gaffer.operation.impl.SetVariable;
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.operation.util.Conditional;
+import uk.gov.gchq.koryphe.impl.function.ToList;
+import uk.gov.gchq.koryphe.impl.function.ToLowerCase;
+import uk.gov.gchq.koryphe.impl.function.ToUpperCase;
+import uk.gov.gchq.koryphe.impl.predicate.IsA;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.ACCUMULO_STORE_SINGLE_USE_PROPERTIES;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.GRAPH_ID_A;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.GRAPH_ID_TEST_FEDERATED_STORE;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.STRING;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.basicEntitySchema;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.contextBlankUser;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.entityBasic;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.loadAccumuloStoreProperties;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.resetForFederatedTests;
+
+public class FederatedStoreInputChainTest {
+    public static final String VAR = "variable";
+    public static final String INPUT_STRING = "AbCd";
+    private FederatedStore federatedStore;
+
+    @AfterAll
+    public static void after() {
+        resetForFederatedTests();
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        resetForFederatedTests();
+
+        federatedStore = new FederatedStore();
+        federatedStore.initialise(GRAPH_ID_TEST_FEDERATED_STORE, null, new FederatedStoreProperties());
+
+        federatedStore.execute(
+                new AddGraph.Builder()
+                        .graphId(GRAPH_ID_A)
+                        .storeProperties(loadAccumuloStoreProperties(ACCUMULO_STORE_SINGLE_USE_PROPERTIES))
+                        .schema(basicEntitySchema())
+                        .isPublic(true)
+                        .build(), contextBlankUser());
+    }
+
+    @Test
+    public void shouldCorrectlySetGenericInputInFederatedOperationChain() throws OperationException {
```

Comment:
Would `shouldCorrectlySetGenericInputInOperationChainWithFederatedOperation` be better? The current name implies that there's a `FederatedOperationChain` - an `OperationChain` which is federated. But that's not what's happening here.

Reply:
It would but I decided this was more concise, will rename